src/devices/HuskyLens/HuskyLens.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Iot.Device.HuskyLens
+{
+    /// <summary>
+    /// todo
+    /// </summary>
+    public class HuskyLens
+    {
+        private readonly IBinaryConnection _connection;
+
+        /// <summary>
+        /// Algorithm currently used.
+        /// </summary>
+        public Algorithm Algorithm { get; private set; }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="connection">guess</param>
+        public HuskyLens(IBinaryConnection connection)
```

Comment:
I would go for something more classical here where you would have 2 constructor, one with SerialPort and one with the details of the serial port (name, etc with default values). It is much simpler from a developer perspective and it hides some internal complexity. 
your IBinaryConnection can then just be internal
Also, is this binding working with something else than serial port? In this case even a stream may be good.

Reply:
HuskyLens also allows I2C, so this was our scaffolding for that. Making this internal seems like a good idea.

For I2C, we would have a `I2cDevice` constructor argument, I think `SerialPort` is at the same level, or is there a `SerialDevice` abstraction?

Reply:
> For I2C, we would have a `I2cDevice` constructor argument

Absolutely


> I think `SerialPort` is at the same level, or is there a `SerialDevice` abstraction

Correct, SerialPort is at the same level, there is no abstraction. So the IBinaryConnection should be internal. Now, some other devices needs that. So it's something that can be moved in internal Common at some point. 
For SerialPort, there are usually 2 constructors, one with the name (and speed + other params if that may be adjusted) and 1 with the SerialPort itself.

Reply:
I'd recommend taking Stream instead of SerialPort as it's easier to plug something else (or even do remote serial port)